### PR TITLE
fix: Grid flexGap lazy check

### LINKED ISSUE #29200: 4.12.0 中新增的 isFlexGapSupported 会导致如果 antd 在 body 前引入会直接挂掉
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

https://github.com/ant-design/ant-design/pull/29141/files/53797cf0581fe41759b8389e3e7e8d7bc005173e#diff-8e9996d165a5fb3da500dcb70b97c1e83a4775d801b1a49934c62322f2302933R33

### Reproduction link
[https://3gpev.csb.app/](https://3gpev.csb.app/)

### Steps to reproduce
在页面的 <body /> 标签前引入 antd 脚本

### What is expected?
不报错

### What is actually happening?
不报错

| Environment | Info |
|---|---|
| antd | 4.12.0 |
| React | * |
| System | * |
| Browser | * |

### LINKED ISSUE #29201: rowStyle中--column-gap及--row-gap值为0时, 参与calc计算时有误
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Steps to reproduce
image
image

### What is expected?
calc表达式中需要有单位才生效, 需要把 0 改为 0px

### What is actually happening?
如图中所示, wrapperCol失效, 图中的输入框原本不会填充满剩余位置

| Environment | Info |
|---|---|
| antd | 4.12.0 |
| React | 17.x |
| System | macOS 11.1 |
| Browser | Chrome 88 |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #29200
fix #29201

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix antd crash when load before page ready.      |
| 🇨🇳 Chinese |   修复 antd 在页面加载之前载入导致的页面崩溃问题。     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/_util/styleChecker.tsx [hunk 1]
```diff
@@ -12,13 +12,16 @@ export const isStyleSupport = (styleName: string | Array<string>): boolean => {
...
-export const isFlexSupported = isStyleSupport(['flex', 'webkitFlex', 'Flex', 'msFlex']);
-
-export const isFlexGapSupported = (() => {
+let flexGapSupported: boolean | undefined;
+export const isFlexGapSupported = () => {
   if (!canUseDocElement()) {
     return false;
   }
 
+  if (flexGapSupported !== undefined) {
```
Comment (shaodahong) [inline components/_util/styleChecker.tsx:21]: 位置可以放到最上面

File: components/_util/styleChecker.tsx [hunk 2]
```diff
@@ -12,13 +12,16 @@ export const isStyleSupport = (styleName: string | Array<string>): boolean => {
   return false;
 };
 
-export const isFlexSupported = isStyleSupport(['flex', 'webkitFlex', 'Flex', 'msFlex']);
-
-export const isFlexGapSupported = (() => {
+let flexGapSupported: boolean | undefined;
+export const isFlexGapSupported = () => {
```
Comment (afc163) [inline components/_util/styleChecker.tsx:16]: 改成 detectFlexGapSupported ？isFlexSupported 是个 boolean，两个类型不同比较奇怪。

File: components/_util/styleChecker.tsx (hunk 1 above)
Comment (zombieJ) [reply components/_util/styleChecker.tsx:21]: 放下面是为了减少 canUseDocElement 分支还得写一个赋值，节省一行代码。

**Comment by afc163** [comment]
> 修复 antd 在页面加载之前载入使用 Grid 导致的页面崩溃问题。

不必使用，umd 包引入即报错。

**Comment by zombieJ** [comment]
rename 一下测试挂了，一脸懵逼……

**Review by shaodahong — APPROVED** [review]

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**